### DISCUSSION TITLE
Filter repos by regex

### DISCUSSION THREAD

**Discussion by nihaals** [community/community#9824] [Other Feature Feedback, Questions, & Ideas] [answered]
If you want to search across a large number of repos, the current options are:
- Filter by org
- Filter by repo (with a lot of `OR`s)

It would be nice if you could also use a regex, which would be especially useful if repo names follow a set format e.g. `<team>-<service name>`. For example, if you wanted to filter by a team, you could do `repo:/^orgname\/teamname-/`.

**Comment by rhaksw** [comment]
Yes, for example when searching across all of github for a domain name [here](https://cs.github.com/?scopeName=All+repos&scope=&q=reveddit+-org%3Areveddit+-org%3Adocile-alligator+-org%3Arhaksw+-org%3Aqaisjp), the repo "darkreader" shows up a lot. There is no way to exclude that from search results. I'd like to be able to say something like `-repo:.*/darkreader`

**Comment by melissachang** [comment]
+1. For my search, there are 50+ (and growing) relevant repos. It's too much work creating 50 `OR`s.

**Comment by GPHemsley-RELX** [comment]
My organization has 7.7k+ repos with 2.6k+ users, all doing different things. To help organize (and because we are migrating from GitLab, which has repo group hierarchy), my team's repos are named with a specific pattern. As soon as I want to look across more than one repo of mine, search becomes effectively unusable because I can't target a wildcard set of repos.

**Comment by Timmmy-nlb** [comment]
There are some news for this topic?
A better search will be very helpfull!

**Comment by brunopini** [comment]
I would +10 this if I could

**Comment by RONNCC** [comment]
Any updates perhaps?

**Comment by LiteBrite82** [comment]
Code search supports regular expressions to search for patterns in your code. You can use regular expressions in bare search terms as well as within many qualifiers, by surrounding the regex in slashes. [GitHub Documentation](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#using-regular-expressions).

Does this cover the specific functionality you are looking for?

**Reply by jfreeland** [reply]
No, that does not cover the specific functionality referred to in this thread.

If you're working in a large organization with many repositories and want to narrow your focus to a specific organization's repos, for instance, you might want to do something like `/repo:hashcorp/sre-.*/` or `/repo:hashicorp/(consul|terraform|vault)/` to exclude other repositories from being included in the results entirely.  At present you would have to do `repo:hashicorp/consul OR repo:hashicorp/terraform OR repo:hashicorp/vault`, as an example, and when you're working with many repositories with a common prefix, it's rather cumbersome to try to do this by hand.

Take for example https://github.com/sourcegraph/sourcegraph/issues/20389.

**Reply by RONNCC** [reply]
An example is we have a naming convention for repos in my company that prepends the business unit to it so i want to do 

something like 

`filetype: python repo:"$orgname-.* " DatabaseWriter`

**Reply by LiteBrite82** [reply]
@look 
Do you have any kung-fu for this?

**Reply by look** [reply, marked as answer]
Unfortunately, no. We'd like to implement this but haven't been able to yet (fun fact, we started on it but ran into some subtle bugs).

**Reply by bric3** [reply]
@look Does it have evolved since the last 8 months ?

**Comment by abiriadev** [comment]
The feature is also helpful in situations such as finding a repository without specifying an organization.

For example, if I want to search for all repositories whose name contains `opencv`, the query should support a format like `repo:*/opencv`. (or at least regex)

**Comment by rneufeldcisco** [comment]
@look Any update on this? It's one of the biggest complaints we have being part of a large organization
